src/taskgraph/util/vcs.py

```diff
@@ -139,6 +153,27 @@ def base_rev(self):
     def branch(self):
         return self.run("branch", "--show-current").strip() or None
 
+    @property
+    def remote_name(self):
+        try:
+            remote_branch_name = self.run(
+                "rev-parse", "--verify", "--abbrev-ref", "--symbolic-full-name", "@{u}"
+            ).strip()
+            return remote_branch_name.split("/")[0]
+        except subprocess.CalledProcessError as e:
+            # Error code 128 comes with the message:
+            # "fatal: no upstream configured for branch $BRANCH"
+            if e.returncode != 128:
+                raise
+
+        remotes = self.run("remote").splitlines()
+        if len(remotes) == 1:
+            return remotes[0]
```

Comment:
Similarly I wonder if we should just use `origin` if it exists?

Reply:
Added.